woocommerce/woocommerce
# CSS3 transitions for all fade animations

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Closes #37691.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add this custom code to your theme's `functions.php` in order to enable fade animations:
```
function carousel_options( $options ) {
	return array_merge(
		$options,
		array(
			'animation' => 'fade',
			'smoothHeight' => false, // smoothHeight is extremely buggy with fade for some reason
			'animationSpeed' => 2600,
			'fadeFirstSlide' => false,
		)
	);
}
add_filter( 'woocommerce_single_product_carousel_options', 'carousel_options' );
```
2. Visit a Single Product Page on a non-touch-enabled device (a PC or a laptop).
3. Confirm that the first gallery image does NOT fadeIn on page load.
4. Confirm that switching between gallery images happens with a fade animation that lasts 2.6 seconds.
5. Enable `fadeFirstSlide` in the custom code in your theme's `functions.php`:
```
function carousel_options( $options ) {
	return array_merge(
		$options,
		array(
			'animation' => 'fade',
			'smoothHeight' => false, // smoothHeight is extremely buggy with fade for some reason
			'animationSpeed' => 2600,
			'fadeFirstSlide' => true,
		)
	);
}
add_filter( 'woocommerce_single_product_carousel_options', 'carousel_options' );
```
6. Visit a Single Product Page on a non-touch-enabled device (a PC or a laptop).
7. Confirm that the first gallery image DOES fadeIn on page load.

`'smoothHeight' => false,` is needed since `smoothHeight` doesn't play nice with `fade` - that's an existing bug with FlexSlider.
